bookwyrm/importers/importer.py

```diff
@@ -17,27 +17,38 @@ class Importer:
     encoding = "UTF-8"
     mandatory_fields = ["Title", "Author"]
 
-    def create_job(self, user, csv_file, include_reviews, privacy):
+    def create_job(self, user, csv_file, job_options, import_options):
         """check over a csv and creates a database entry for the job"""
         job = ImportJob.objects.create(
-            user=user, include_reviews=include_reviews, privacy=privacy
+            user=user,
+            include_reviews=job_options["include_reviews"],
+            privacy=job_options["privacy"],
         )
-        for index, entry in enumerate(
-            list(csv.DictReader(csv_file, delimiter=self.delimiter))
-        ):
-            if not all(x in entry for x in self.mandatory_fields):
-                raise ValueError("Author and title must be in data.")
-            entry = self.parse_fields(entry)
-            self.save_item(job, index, entry)
+        if import_options["file_type"] == "txt":
+            for index, entry in enumerate(csv_file.read().splitlines()):
+                entry = self.parse_fields(
+                    entry, default_shelf=import_options["default_shelf"]
+                )
+                self.save_item(job, index, entry)
+        else:
+            for index, entry in enumerate(
+                list(csv.DictReader(csv_file, delimiter=self.delimiter))
+            ):
+                if not all(x in entry for x in self.mandatory_fields):
+                    raise ValueError("Author and title must be in data.")
+                entry = self.parse_fields(entry)
+                self.save_item(job, index, entry)
         return job
 
     def save_item(self, job, index, data):  # pylint: disable=no-self-use
         """creates and saves an import item"""
         ImportItem(job=job, index=index, data=data).save()
 
-    def parse_fields(self, entry):
+    def parse_fields(self, entry, default_shelf=None):
         """updates csv data with additional info"""
         entry.update({"import_source": self.service})
+        if default_shelf:
+            entry.update({"Exclusive Shelf": default_shelf})
```

Comment:
We can now override the shelf for every importer, but it's only enabled for `txt` files.

Reply:
> The last commit fails automated tests:
> ...
> But the test works on a local setting.

Usually when this happens, it means the local tests are making a live http request or using a service that isn't running in the automated test environment. I'd guess that it's doing a lookup to openlibrary or inventaire to resolve the import items locally, which won't work in automated tests. If that's what's happening, you can use the `responses` library to mock GET requests.

Reply:
Ok this is weird, the test file has been reverted to its original content, but the test still turns out failing. From what I understand, `test_create_job` doesn't try to make a live http request, it just tests that the job is created, and that each imported item exists. Could the error be caused by the exectution of another test earlier, and their presence in the db? (I'm out of my depth, I might have overlooked something)